added codes for extracting AAC features for 7 and 8 classes

HamidHeyde opened the pull request.

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+        aac1=open((os.getcwd()+destination1), 'a')
+        aac1.write((str(aminoLetters).replace('[','')).replace(']','')+",Label")
+
+        log("Extraicting AAC from "+dataset+" for 7 classes")
```
typo

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+#Rounds Floating Points Number
+#============================
+def roundFloatingpoint(num):
+  return (np.round(num*1000)/1000)
```
Why not just round to the 3rd decimal? https://www.w3schools.com/python/ref_func_round.asp
Also, you shouldn't require numpy for just that.

HamidHeyde replied:
I've included Numpy for some other functionality(in classification part). But i used it for this problem as well. because, i found a webSite online which would produce the same feature for a Sequence, and the only way to produce the same result was using the Numpy.round() function.

Round() function generates some results which is really close, but not the same thing. So, to stay on the safe side, i used the numpy version instead of the other one.

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+#Removes Undesired Character
+#============================
+def removeSpaceNewLine(seq):
+  return ((seq.replace('\n','')).replace(' ',''))
```
Non portable

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+  counter=0
+  for alphabet in seq:
+    if (alphabet==letter):
+      counter=counter+1
```
You should use the `count` function of list for that: https://stackoverflow.com/questions/2600191/how-can-i-count-the-occurrences-of-a-list-item

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+        open((os.getcwd()+fileAddress), 'w').close()
+    except:
+        newPath=(fileAddress).split("/")
+        path = ("/".join(newPath[:(len(newPath)-1)]))
```
non portable

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+    try:
+        # Reading Source Transporter
+        sourceHandler=open(source,'r')
+        sourceData=(sourceHandler.read()).split("\n")
```
non portable

glatard commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+        for line in sourceData:
+            if '>' not in line:
+                #Calculating the AAC feature and Writes it to aac.csv
+                destination.write('\n'+
```
non portable (check all occurrences of `+`, `/` and `\n` in string manipulations)

HamidHeyde commented on helpers.py:
```diff
@@ -102,3 +102,152 @@ def downloadCreateDataList(itemList,site,format,dataPath):
+                nonTransporters.append(file[:-6])
+            else:
+                if "fasta" in file:
+                    trasnporters.append(file[:-6])
```
I modified those using "list comprehension" but then i had to run the same loop twice. once for transporters and once for non-transporters. so i left it as its was before. The list comprehension version is as follow:

# trasnporters = [file[:-6] for file in os.listdir(os.getcwd()+source) if "nonTransporter" not in file]
# nonTransporters = [file[:-6] for file in os.listdir(os.getcwd()+source) if "nonTransporter" in file]

HamidHeyde commented:
Hi @glatard,

I modified the codes, so, it could be run on other platforms too. So,

- I Removed the config.json.
- Moved the config to helpers.py.
- and also applied the necessary modifications to all 3 python files (download.py, extractFeature.py, helpers.py) accordingly.

Best Regards,
Hamid
